infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInfrastructure.java

```diff
@@ -81,6 +88,17 @@ public boolean isNamespaceValid(String name) {
     return NamespaceNameValidator.isValid(name);
   }
 
+  @Override
+  public Response sendDirectInfrastructureRequest(
+      String httpMethod, URI relativeUri, @Nullable JsonNode body) throws InfrastructureException {
```

Comment:
What do you think about `JsonNode body -> String body` or Reader body?

Reply:
`JsonNode body` -> @HttpHeaders headers, InputStream body`